tests/php/Forms/FormFieldTest.php

```diff
@@ -455,6 +474,99 @@ public function testGetSchemaStateWithFormValidation()
+        $field->setTriggerTestValidationError(true);
+
+        // Ensure messages set via updateValidationResult() propagate through to form fields after validation
+        $form = new Form(null, 'TestForm', new FieldList($field), new FieldList(), new RequiredFields());
+        $form->validationResult();
+        $schema = $field->getSchemaState();
+        $this->assertEquals(
+            'A test error message',
+            $schema['message']['value']
+        );
+    }
+
+    public function testValidationExtensionHooksAreCalledOnFormFieldSubclasses()
+    {
+        // Can't use a provider for this due to class manifest weirdness
+        $formFieldClasses = ClassInfo::subclassesFor(FormField::class, false);
+        foreach ($formFieldClasses as $formFieldClass) {
+            $reflection = new ReflectionClass($formFieldClass);
+            // Skip abstract classes like MultiSelectField
+            if ($reflection->isAbstract()) {
+                continue;
+            }
+
+            switch ($formFieldClass) {
+                case NullableField::class:
+                case CompositeField::class:
+                case FieldGroup::class:
+                case PopoverField::class:
+                    $args = [TextField::create('Test2')];
+                    break;
+                case SelectionGroup_Item::class:
+                    $args = ['Test', [TextField::create('Test2')]];
+                    break;
+                case ToggleCompositeField::class:
+                    $args = ['Test', 'Test', TextField::create('Test2')];
+                    break;
+                case PrintableTransformation_TabSet::class:
+                    $args = [Tab::create('TestTab', 'Testtab', TextField::create('Test2'))];
+                    break;
+                case TreeDropdownField::class:
+                case TreeDropdownField_Readonly::class:
+                    $args = ['Test', 'Test', Group::class];
+                    break;
+                case PermissionCheckboxSetField::class:
+                case PermissionCheckboxSetField_Readonly::class:
+                    $args = ['Test', 'Test', Permission::class, 'Test'];
+                    break;
+                case SelectionGroup::class:
+                    $args = ['Test', []];
+                    break;
+                case GridField_FormAction::class:
+                    $args = [GridField::create('GF'), 'Test', 'Test label', 'Test action name', []];
+                    break;
+                case GridState::class:
+                    $args = [GridField::create('GF')];
+                    break;
+                default:
+                    $args = ['Test', 'Test label'];
+            }
+
+            echo $formFieldClass . "\n";
+            ob_flush();
+
+            $mock = $this->getMockBuilder($formFieldClass)
+                ->setConstructorArgs($args)
+                ->setMethods(['extendValidationResult'])
```

Comment:
```suggestion
                ->onlyMethods(['extendValidationResult'])
```
Seems `setMethods()` is deprecated.